exchange/index.go

```diff
@@ -105,21 +113,29 @@ func WithUpdateFunc(fn func()) IndexOption {
 	}
 }
 
+// UpdateOptions takes IndexOption parameters to change Index behavior
+func (idx *Index) UpdateOptions(opts ...IndexOption) {
```

Comment:
Not a fan of creating a method for updating options during the tests. I don't think you actually need it.